sdk/objectanchors/Azure.MixedReality.ObjectAnchors.Conversion/src/NotSupportedAssetFileTypeException.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.Serialization;
+
+namespace Azure.MixedReality.ObjectAnchors.Conversion
+{
+    /// <summary>
+    /// An Exception thrown during an attempt to provide an unsupported asset file type in an asset conversion operation
+    /// </summary>
+    public class NotSupportedAssetFileTypeException : Exception, ISerializable
+    {
+        /// <summary>
+        /// Creates an instance of the UnsupportedAssetFileTypeException
+        /// </summary>
+        public NotSupportedAssetFileTypeException()
+            : base($"The provided asset file type is unsupported by Azure Object Anchors for conversion.")
+        {
+        }
+
+        /// <summary>
+        /// Creates an instance of the UnsupportedAssetFileTypeException
+        /// </summary>
+        /// <param name="message">The message corresponding to the exception</param>
+        public NotSupportedAssetFileTypeException(string message)
+            : base(message)
+        {
+        }
+
+        /// <summary>
+        /// An Exception thrown during an attempt to provide an unsupported asset file type in an asset conversion operation
+        /// </summary>
+        /// <param name="message">The message corresponding to the exception</param>
+        /// <param name="inner">The inner exception</param>
+        public NotSupportedAssetFileTypeException(string message, Exception inner)
+            : base(message, inner)
+        {
+        }
+
+        internal NotSupportedAssetFileTypeException(AssetFileType assetFileType, IEnumerable<AssetFileType> supportedAssetFileTypes)
+            : base($"The provided asset file type of \"{assetFileType}\" is unsupported by Azure Object Anchors for conversion. Supported file types: {string.Join(", ", supportedAssetFileTypes)}")
+        {
+            AttemptedFileType = assetFileType;
+            SupportedAssetFileTypes = supportedAssetFileTypes;
+        }
+
+        /// <summary>
+        /// An Exception thrown during an attempt to provide an unsupported asset file type in an asset conversion operation
+        /// </summary>
+        /// <param name="info">The SerializationInfo</param>
+        /// <param name="context">The StreamingContext</param>
+        protected NotSupportedAssetFileTypeException(SerializationInfo info, StreamingContext context)
```

Comment:
Do we need this? it looks like it is just calling through to base ctor.

Reply:
See section 7.4 of the FXDG

Reply:
(But it'd be worth adding a comment that this is required for serialization so no one takes it out in the future.)

Reply:
Yeah but there is no custom implementation here, so why not just let the Exception constructor handle it?

Reply:
Or we should implement GetObjectData to include the additional data that we want to serialize.

Reply:
It would be hidden:

```C#
public class Foo
{
    public Foo() : this(0) { }
    protected Foo(int x) => X = x;
    public int X { get; }
}

public class Bar : Foo
{
    public Bar() : base() { }
}

public class Baz : Bar
{
    public Baz() : base() { }
    protected Baz(int x) : base(x) { }  // error CS1729: 'Bar' does not contain a constructor that takes 1 arguments
}
```

Reply:
If someone were to inherit, it would be hidden, but we are not doing anything in the ctor now so why would it matter?

Reply:
Ah nevermind, I guess that makes sense - if someone wanted to create a custom exception that inherits from this type and add custom serialization, they would need this ctor.

Reply:
We should probably consider adding analyzers for this as I don't think we have been following this guideline everywhere.

Reply:
We just need to turn on https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1032 if it's not already running today.

Reply:
But still, should we add the existing properties into the serialization info, e.g. https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/RequestFailedException.cs#L78

Reply:
Sure, that would be a good thing for the two new members.

Reply:
CA1032 is disabled for some reason - 

https://github.com/Azure/azure-sdk-for-net/blob/4162f6fa2445b2127468b9cfd080f01c9da88eba/eng/Directory.Build.Common.props#L38

Serialization warning is not disabled, yet it is not firing for ServiceBusException or EventHubsException - https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2229

Reply:
Ah it is not documented, but the warning only fires if you have Serializable attribute added to your type.

Reply:
@tg-msft thoughts here? this was based on your feedback

Reply:
Yes, see above where I agreed with Josh that we should add those two new members in the serialization .ctor like the example he linked.

Reply:
Addressed in a similar manner